WPMedia/fusion-news-theme-blocks
# Tmedia 572 show email editable field

## Description
Show editable email field on profile

## Jira Ticket
- [TMEDIA-572](https://arcpublishing.atlassian.net/browse/TMEDIA-572)

## Acceptance Criteria
Add custom field to block (TMEDIA-578) to enable editing Email address

If unchecked no email address management is shown

If checked:

Show email address edit functionality as detailed below

Show “Account Management” heading string (translated)

Use existing input and button components

Show “Email Address” string (translated)

Current email address

Clicking “Edit”

Shows email address field with current email address

User can edit email address

Email address is validated against rules already in place

If user uses the “cancel” button - 

user is returned to previous state

current email address is shown with edit button to right side

If user uses the “save” button

Make API call to save Users email address is saved to their account

If successful return to initial state

If an error occurs show an error message

Email is displayed when not in “Edit” state

Edit link opens editable email field

Invalid email shows the error state

## Test Steps

1. Checkout this branch `git checkout TMEDIA-572-show-email-editable-field`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/identity-block`
3. Create a page with account management
4. See no email input shown by default 
5. Elect to show enable email 
6. Publish page
7. Login to a user 
8. Go to account management page 
9. See your current email designed to mocks 
10. Click the edit button 
11. See the input and your current email
12. Make changes in the input
13. Click cancel. You should see your initial values set -- not your changed text
14. Click edit again.
15. Change values and click submit. Your new email should be set and visible. 

Can also see and play with the reusable component via `npm run storybook`

## Effect Of Changes
### Before
- profile page did not show email

### After
- profile page shows email and allows you to edit it 

## Dependencies or Side Effects
- Looks like there was other dependencies around email not set but this seemed like a typo or deferred to social

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
